### PR TITLE
Support repeats with correct indexing

### PR DESCRIPTION
**Pull request by SCdF**
medic/medic-webapp#4248

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: static/js/enketo/widgets/db-object-widget.js [hunk 1]
```diff
@@ -81,12 +81,14 @@ define( function( require, exports, module ) {
         var doc = selected && selected[0] && selected[0].doc;
         if (doc) {
             var field = $this.attr('name');
+            var domIndex = $('[name="' + field + '"]').index(this);
+            var xmlIndex = domIndex + 1;
```
Comment (SCdF) [inline static/js/enketo/widgets/db-object-widget.js]: Yes, XSLT indexes from 1. I don't know how those people sleep at night.

File: static/js/enketo/widgets/db-object-widget.js [hunk 2]
```diff
@@ -102,9 +104,11 @@ define( function( require, exports, module ) {
             }
             if (_.isObject(value)) {
                 // recursively set fields for children
-                return updateFields(value, path, originatingKeyPath);
+                return updateFields(value, path, index, originatingKeyPath);
             }
-            var node = Enketo.getCurrentForm().model.node(path);
+
+            var node = Enketo.getCurrentForm().model.node('(' + path + ')[' + index + ']');
```
Comment (SCdF) [inline static/js/enketo/widgets/db-object-widget.js]: This looks like it's more precedence than you need, but it's not! `[]` has higher precedence than paths, and so if you don't do this weird stuff happens that I don't logically understand, except as much as to understand that it's wrong.

Comment (garethbowen) [reply static/js/enketo/widgets/db-object-widget.js]: Did you know, the [node](https://github.com/enketo/enketo-core/blob/master/src/js/Form-model.js#L250) function takes an `index` parameter? I don't know if it'll do what you need but it's worth a try!

Comment (SCdF) [reply static/js/enketo/widgets/db-object-widget.js]: Nice tip! It is also zero-indexed like God intended.

**Pull request merged**